napari/layers/base/_base_mouse_bindings.py

```diff
@@ -155,9 +167,9 @@ def transform_with_box(layer, event):
 
     # we work in data space so we're axis aligned which simplifies calculation
     # same as Layer.data_to_world
-    initial_data_to_world = layer._transforms[1:].simplified.set_slice(
-        event.dims_displayed
-    )
+    simplified = layer._transforms[1:].simplified
+    assert simplified is not None
```

Comment:
Sorry, naive question: I don't get this assert vs. if/else or even try?

Reply:
The problem is that for transform chain simplified may return none if transform is empty. 

Personally I prefer to change logic of transform and raise exception when called simplified on empty transform chain

Reply:
AH, so by asserting we make the type checker happy with the return type?

Reply:
Yep, that's right

Reply:
I prefer to merge #6674 first and remove this assert